osf/migrations/0155_add_registration_files_count.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2019-01-18 14:43
+from __future__ import unicode_literals
+import logging
+
+from django.db import migrations, models
+from django_bulk_update.helper import bulk_update
+
+logger = logging.getLogger(__file__)
+
+def add_registration_files_count(state, *args, **kwargs):
+    AbstractNode = state.get_model('osf.abstractnode')
+    BaseFileNode = state.get_model('osf.basefilenode')
+    ContentType = state.get_model('contenttypes', 'ContentType')
+    registrations = AbstractNode.objects.filter(type='osf.registration', is_deleted=False)
+    content_type = ContentType.objects.get(app_label='osf', model='abstractnode')
+
+    registrations_to_update = []
+    for registration in registrations:
+        job = registration.archive_jobs.first() if registration.archive_jobs.count() else None
+        archiving = job and not job.done and (job.status != 'SUCCESS')
+        if archiving:
+            # Skip stuck, failed, or archiving registrations.
+            continue
+        registration_files = BaseFileNode.active.filter(
```

Comment:
I got an error here:

```
Traceback (most recent call last):
  File "manage.py", line 21, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/sloria/.pyenv/versions/osf/lib/python2.7/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/Users/sloria/.pyenv/versions/osf/lib/python2.7/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/sloria/.pyenv/versions/osf/lib/python2.7/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/sloria/.pyenv/versions/osf/lib/python2.7/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/Users/sloria/.pyenv/versions/osf/lib/python2.7/site-packages/django/core/management/commands/migrate.py", line 204, in handle
    fake_initial=fake_initial,
  File "/Users/sloria/.pyenv/versions/osf/lib/python2.7/site-packages/django/db/migrations/executor.py", line 115, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/Users/sloria/.pyenv/versions/osf/lib/python2.7/site-packages/django/db/migrations/executor.py", line 145, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/Users/sloria/.pyenv/versions/osf/lib/python2.7/site-packages/django/db/migrations/executor.py", line 244, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/Users/sloria/.pyenv/versions/osf/lib/python2.7/site-packages/django/db/migrations/migration.py", line 129, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/Users/sloria/.pyenv/versions/osf/lib/python2.7/site-packages/django/db/migrations/operations/special.py", line 193, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/Users/sloria/projects/cos-projects/osf/osf/migrations/0155_add_registration_files_count.py", line 25, in add_registration_files_count
    registration_files = BaseFileNode.active.filter(
AttributeError: type object 'BaseFileNode' has no attribute 'active'
```

I think this is because `BaseFileNode` is being loaded with `state.get_model`, which will load the class *without custom managers*. So you'll need to do the query "manually" using BaseFileNode.objects.filter(...).